### PR TITLE
[ADD] product_gs1_barcode and encoding (base_gs1_barcode)

### PR DESCRIPTION
**Pull request by olgamarcocb**
Cherry-pick (#386) and Migration to 14.0

### CODE REVIEW

File: base_gs1_barcode/models/gs1_encode_barcode.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Copyright 2021 Creu Blanca
+
+from datetime import date, datetime
+
+from odoo import _, exceptions, models
+
+
+class GS1Barcode(models.AbstractModel):
```
Comment (bealdav) [inline base_gs1_barcode/models/gs1_encode_barcode.py]:
```suggestion
class GS1EncodeBarcode(models.AbstractModel):
```

File: product_gs1_barcode/models/gs1_encode_barcode.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        return result
+
+
+class StockProductionLot(models.Model):
+    _name = "stock.production.lot"
+    _inherit = ["stock.production.lot", "gs1.encode.barcode"]
+
+    def _encode_gs1_vals(self):
+        result = self.product_id._encode_gs1_vals()
+        result.update(super(StockProductionLot, self)._encode_gs1_vals())
```
Comment (bealdav) [inline product_gs1_barcode/models/gs1_encode_barcode.py]:
```suggestion
        result.update(super()._encode_gs1_vals())
```

**Review by bealdav — APPROVED** [review]

**Review by etobella — APPROVED** [review]

**Review by ikapasi-initos — APPROVED** [review]

**Review by ikapasi-initos — COMMENTED** [review]
Code review and functional test
LGTM

**Pull request merged**